packages/pyright-internal/src/common/nodeWorkersHost.ts

```diff
@@ -0,0 +1,88 @@
+import {
+    parentPort,
+    MessagePort as WorkerThreadsMessagePort,
+    Worker as WorkerThreadsWorker,
+    MessageChannel as WorkerThreadsMessageChannel,
+    threadId,
+} from 'worker_threads';
+import {
+    MessageChannel,
+    MessagePort,
+    MessageSourceSink,
+    shallowReplace,
+    Transferable,
+    WorkersHost,
+} from './workersHost';
+
+export class NodeWorkersHost implements WorkersHost {
```

Comment:
I have never run this code so need to find a reasonable way to test it.

The tests that I can run in the internal project don't really exercise the higher-level features.

Reply:
All seems to be working.

I was considerably confused by the fact that the background analysis is explicitly disabled in debug mode, but hacking around that all is well.